diff --git a/rust/Makefile b/rust/Makefile
index 7842ad0a4ea7..134d52411884 100644
--- a/rust/Makefile
+++ b/rust/Makefile
@@ -331,12 +331,6 @@ bindgen_c_flags = $(filter-out $(bindgen_skip_c_flags), $(c_flags)) \
 	$(bindgen_extra_c_flags)
 endif
 
-ifdef CONFIG_LTO
-bindgen_c_flags_lto = $(filter-out $(CC_FLAGS_LTO), $(bindgen_c_flags))
-else
-bindgen_c_flags_lto = $(bindgen_c_flags)
-endif
-
 # `-fno-builtin` is passed to avoid `bindgen` from using `clang` builtin
 # prototypes for functions like `memcpy` -- if this flag is not passed,
 # `bindgen`-generated prototypes use `c_ulong` or `c_uint` depending on
