diff --git a/arch/x86/Kconfig.cpu b/arch/x86/Kconfig.cpu
index 96750ac83770..084513894ede 100644
--- a/arch/x86/Kconfig.cpu
+++ b/arch/x86/Kconfig.cpu
@@ -243,6 +243,15 @@ config MATOM
 	  accordingly optimized code. Use a recent GCC with specific Atom
 	  support in order to fully benefit from selecting this option.
 
+config MZEN3
+	bool "AMD Ryzen 3"
+	depends on X86_64
+	depends on (CC_IS_GCC && GCC_VERSION >= 100300) || (CC_IS_CLANG && CLANG_VERSION >= 120000)
+	help
+	  Select this for AMD Family 19h Zen 3 processors.
+
+	  Enables -march=znver3
+
 endchoice
 
 config CC_HAS_MARCH_NATIVE
diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 5bae3230b32d..e257af02ff57 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -176,6 +176,10 @@ else
 	KBUILD_RUSTFLAGS += -Ctarget-cpu=x86-64-v$(CONFIG_X86_64_ISA_LEVEL)
     endif
 endif
+ifdef CONFIG_MZEN3
+        KBUILD_CFLAGS += -march=znver3
+        KBUILD_RUSTFLAGS += -Ctarget-cpu=znver3
+endif
 
         KBUILD_CFLAGS += -mno-red-zone
         KBUILD_CFLAGS += -mcmodel=kernel
