diff --git a/cross-wine.conf b/cross-wine.conf
index 3e3301ac..de72245e 100644
--- a/cross-wine.conf
+++ b/cross-wine.conf
@@ -23,6 +23,6 @@ needs_exe_wrapper = true
 cpp_link_args = ['-mwindows']
 
 # For instance, this should be in the 64-bit only cross-file
-# c_args = ['-m64']
-# cpp_args = ['-m64']
-# cpp_link_args = ['-m64', '-mwindows']
+c_args = ['-m64']
+cpp_args = ['-m64']
+cpp_link_args = ['-m64', '-mwindows']
